src/user/proposals/ProposalDiscussion/DiscussionContent/DiscussionContent.js

```diff
@@ -1,24 +1,104 @@
 import React, { Component } from "react";
 import "./DiscussionContent.scss";
-import { Button, Container, Row, Col, Image, ListGroup } from "react-bootstrap";
+import { Button, Badge, Image, ListGroup } from "react-bootstrap";
 import DiscussionComments from "./DiscussionComments/DiscussionComments";
 import eventImg from "../../../../svgs/event-img-1.svg";
 import userIcon2 from "../../../../images/userIcon2.jpg";
 import RequestChanges from "../DiscussionPopups/RequestChanges";
+import { withRouter, Link } from "react-router-dom";
+import { Editor } from "@tinymce/tinymce-react";
+import Carousel, { Modal, ModalGateway } from "react-images";
 
 class DiscussionContent extends Component {
   constructor(props) {
     super(props);
-    this.state = { comments: [], showModal: false, selectedText: "" };
+    this.state = {
+      comments: [],
+      showModal: false,
+      selectedText: "",
+      proposalId: "",
+      isAdmin: false,
+      proposalTitle: "",
+      markdownString: "",
+      proposalDescription: "",
+      commentList: [],
+      author: "",
+      images: [{ source: "../../../../images/userIcon2.jpg" }],
+      imageModalOpen: false,
+      proposalState: "",
+      variant: "danger",
+    };
   }
 
+  //Token would be passed down from the
   componentDidMount() {
-    this.processComments();
+    const { proposalId, isAdmin, userId, token } = this.props.location.state;
+
+    this.setState(
+      {
+        proposalId: proposalId,
+        isAdmin: isAdmin,
+        userId: userId,
+        token: token,
+      },
+      () => {
+        console.log(this.state);
+        this.getData();
+      }
+    );
   }
+
+  getData = () => {
+    console.log("getDtata called");
+    fetch("http://localhost:5000/proposal/" + this.state.proposalId, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: this.state.token,
```

Comment:
No need to configure fetch or axios every time, just configure once for all the requests to the server